app/tiebout.hoon

```diff
@@ -0,0 +1,236 @@
+  ^-  (quip move _this)
+  [~ this(baseurl.sta burl)]
+::
+::  +send-notify: if king, send hiss. if not, do nothing.
+::
+++  send-notify
+  |=  not=notification
+  ^-  (quip move _this)
+  ?:  =(king.sta our.bol)
+    :_  this
+    [ost.bol %hiss /request [~ ~] %httr %hiss (create-apns-request not)]~
+  [~ this]
+::
+::  +diff-hall-prize: receive new circle data
+::
+++  diff-hall-prize
+  |=  [wir=wire piz=prize:hall]
+  ^-  (quip move _this)
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ::
+  ::  %our: set config of circle and iterate through messages, sending
+  ::  notifications for all messages where number is higher than our last-read
+  ::
+      {%our @ @}
+    ?>  ?=(%circle -.piz)
+    =/  nom/name:hall  i.t.wir
+    =/  red/@ud  red.loc.cos.piz
+    [~ this(circles.sta (~(put by circles.sta) nom red))]
+  ==
+::
+::  +reap: recieve acknowledgement for peer
+::
+++  reap
+  |=  [wir=wire err=(unit tang)]
+  ^-  (quip move _this)
+  ?~  err
+    [~ this]
+  ?~  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ?+  i.wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+    %our
+    ?<  ?=(~ t.wir)
+    [~ this]
+  ==
+::
+::  +quit: receive subscription failed, resubscribe
+::
+++  quit
+  |=  wir=wire
+  ^-  (quip move _this)
+  ?~  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ?+  i.wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+      %our
+    ?<  ?=(~ t.wir)
+    :_  this
+    =/  doc/dock  [our.bol %hall]
+    [ost.bol %peer /our/[i.t.wir] doc /circle/[i.t.wir]/config/grams]~
+  ==
+
+::
+::  +diff-hall-rumor: receive message or a read event from a hall circle
+::
+++  diff-hall-rumor
+  |=  [wir=wire rum=rumor:hall]
+  ^-  (quip move _this)
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ::
+  ::  %our
+  ::
+      {%our @ @}
+    ?>  ?=(%circle -.rum)
+    =/  nom/name:hall  i.t.wir
+    ?+  -.rum.rum
+      [~ this]
+    ::
+    ::  %gram: send notification if envelope is lower than read number
+    ::
+      %gram
+    =/  red  (~(get by circles.sta) nom)
+    ?~  red
+      (mean [leaf+"invalid circle for diff: {(spud wir)}"]~)
+    ?:  (gth num.nev.rum.rum u.red)
+      :_  this(circles.sta (~(put by circles.sta) nom u.red))
+      (conditional-msg-to-not u.red nev.rum.rum)
+    :_  this
+    (conditional-msg-to-not u.red nev.rum.rum)
+    ::
+    ::   %config: set our read number
+    ::
+      %config
+    ?+  -.dif.rum.rum
+      [~ this]
+      ::
+      %read
+    [~ this(circles.sta (~(put by circles.sta) nom red.dif.rum.rum))]
+    ==
+    ==
+  ==
+::
+::  generate notification move from hall message if conditions are met
+::
+++  conditional-msg-to-not
+  |=  [red=@ud env=envelope:hall]
+  ^-  (list move)
+  ?:  =(aut.gam.env our.bol)
+    ~
+  =/  pay  %-  my  :~
+    alert+s+'New message from {(cite:title aut.gam.env)}'
+  ==
+  =/  not/notification  :+
+    token.sta
+    'com.tlon.urbit-client'
+    pay
```

Comment:
You're probably fine putting these between `[]` on one line. If you feel that's too long, proper `:+` form would be

```hoon
:+  token.sta
  'com.tlon.urbit-client'
pay
```